File: ios/Plugin/CapacitorMLKitPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(CapacitorMLKitPlugin, "CapacitorMLKit",
           CAP_PLUGIN_METHOD(echo, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(ocr, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(barcode, CAPPluginReturnPromise);
)

File: ios/Plugin/CapacitorMLKitPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(CapacitorMLKitPlugin)
public class CapacitorMLKitPlugin: CAPPlugin {
    private let implementation = CapacitorMLKit()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
    
    @objc func ocr(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            self.implementation.ocr(call)
        }
    }
    
    @objc func barcode(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            self.implementation.barcode(call)
        }
    }
    
   
    
}
